Feature/kas 3106 vertalingen herwerken

TaneeG opened the pull request:
jira: https://kanselarij.atlassian.net/browse/KAS-3106

styling wordt vrijdag door @brenner-company bekeken

Commit:
KAS-3097 Delete index and rows & rework model

Commit:
KAS-3097 Getting translationRows to load requests

Commit:
Merge branch 'development' into feature/KAS-3106_Vertalingen_herwerken

Commit:
KAS-3106 Add panels & remove some unused code

Commit:
Merge branch 'development' into feature/KAS-3106_Vertalingen_herwerken

Commit:
Merge branch 'development' into feature/KAS-3106_Vertalingen_herwerken

# Conflicts:
#	app/pods/publications/publication/translations/controller.js
#	app/pods/publications/publication/translations/template.hbs
#	cypress/selectors/publication.selectors.js

Commit:
KAS-3106 REwork the new flow for requesting & uploading translations

Commit:
Merge branch 'development' into feature/KAS-3106_Vertalingen_herwerken

Commit:
Merge branch 'development' into feature/KAS-3106_Vertalingen_herwerken

Commit:
KAS-3106 Add delete request

Commit:
Merge branch 'development' into feature/KAS-3106_Vertalingen_herwerken

# Conflicts:
#	app/pods/publications/publication/translations/documents/controller.js

Commit:
KAS-3106 remove unused import

Commit:
Merge branch 'development' into feature/KAS-3106_Vertalingen_herwerken

Commit:
KAS-3106 remove routes

erikap commented on app/pods/publications/publication/translations/route.js:
```diff
@@ -1,16 +1,86 @@
+// use:
+// - isPieceDeletable property
+// - file property avoids error when piece (and file) are deleted
+export class TranslationRow {
+  @tracked piece;
+  @tracked date;
+  @tracked requestActivity;
+
+  // no async constructor() in JS
+  static async create(piece,requestActivity) {
+    const row = new TranslationRow();
+
+    if (piece){
+      row.piece = piece;
+      row.date = piece.created;
+    }
+    if (requestActivity) {
+      row.requestActivity = requestActivity;
+      row.date = requestActivity.startDate;
+    }
+    return row;
+  }
+
+  get isPieceDeletable() {
+    // can be translation or publication related
+    const isUsedInRequest = this.requestActivitiesUsedBy.length > 0;
+    // receivedDate is set if and only if it is a received pieced
+    const isReceived = !!this.piece.receivedDate;
+    const isUsed = isUsedInRequest || isReceived;
+    return !isUsed;
+  }
+}
```
Heeft deze class nog een meerwaarde?  In het vorige design was deze class nodig om de tabel met documenten op een eenvoudige manier te kunnen opbouwen in de template, maar aangezien het design nu dichter bij het model aanleunt, zie ik de meerwaarde niet. 
De property `isPieceDeletable` wordt onder vertaling bijvoorbeeld niet meer gebruikt voor zover ik zie.

erikap commented on app/pods/publications/publication/translations/route.js:
```diff
@@ -1,16 +1,86 @@
 export default class PublicationsPublicationTranslationsRoute extends Route {
-  model() {
-    return this.modelFor('publications.publication').translationSubcase;
+  async model() {
+    this.translationSubcase = await this.modelFor('publications.publication').translationSubcase;
+
+    let requestActivities = await this.store.query('request-activity',
+      {
+        'filter[translation-subcase][:id:]': this.translationSubcase.id,
+        include: 'translation-activity,email,used-pieces,used-pieces.file',
+        sort: '-start-date',
+      }
+    );
+    let requestRows = await Promise.all(requestActivities.map((request) => TranslationRow.create(null,request)));
+
+    // Received translation documents generated by a translation-activity
+    const pieces = await this.store.query('piece', {
+      'filter[translation-activity-generated-by][subcase][:id:]': this.translationSubcase.id,
+      include: [
+        'file',
+        'publication-subcase-source-for',
+      ].join(','),
+      'page[size]': PAGE_SIZE.PUBLICATION_FLOW_PIECES,
+    });
+
+    const pieceRows = await Promise.all(pieces.map((piece) => TranslationRow.create(piece,null)));
+
+    let translationRows= [pieceRows,requestRows];
+    translationRows = translationRows.flatMap((row) => row.toArray())
+    translationRows = translationRows.sortBy('date').reverseObjects();
+
+    return translationRows;
```
Deze model hook is niet logisch. Als je naar het design kijkt, bevat het scherm een tijdslijn van aanvraag-activiteiten en vertaal-activiteiten. Dat is dan ook wat ik zou verwachten in de model-hook, maar het is hier een mix van aanvraag-activiteit en pieces, die dan nog eens gewrapped zijn in een custom class.

erikap commented on app/pods/publications/publication/translations/template.hbs:
```diff
@@ -1,23 +1,82 @@
-{{outlet}}
\ No newline at end of file
+<div class="auk-u-m-4">
+  <Publications::Publication::Translations::TranslationsInfoPanel
+    @translationSubcase={{this.translationSubcase}}
+    @isFinal={{this.publicationFlow.status.isFinal}}
+  />
```
Kan je deze route nog verder opsplitsen zodat het info panel op `publications.publication.translations` staan en de activiteiten-timeline op de index subroute `publications.publication.translations.index` komt te staan. Op die manier gaan de model-hooks cleaner worden. 
Die structuur is nu ook gebruikt voor het info-panel vs de referentiedocumenten op de 'Beslissingen'-tab. Zie https://github.com/kanselarij-vlaanderen/frontend-kaleidos/tree/development/app/pods/publications/publication/decisions

erikap commented on app/pods/publications/publication/translations/controller.js:
```diff
@@ -0,0 +1,197 @@
+      language: language,
+      translationActivityGeneratedBy: translationActivity,
+    });
+    if (translationUpload.isSourceForProofPrint) {
+      piece.publicationSubcaseSourceFor = this.publicationSubcase;
+    }
```
Niet meer relevant? De property komt op nog plaatsen voor in components, maar wordt nergens nog getoond in de UI. Moet dus opgekuist worden.

erikap commented on app/pods/publications/publication/translations/template.hbs:
```diff
@@ -1,23 +1,82 @@
+        {{#each this.model as |translationRow|}}
+          {{#if translationRow.requestActivity}}
+            <Publications::Publication::Translations::TranslationRequestedPanel
+              @translationSubcase={{this.translationSubcase}}
```
Dit argument lijkt niet gebruikt in de component

erikap commented on app/components/publications/publication/translations/translation-requested-panel.js:
```diff
@@ -0,0 +1,35 @@
+  @tracked isVerifyingDelete = false;
+
+
+  constructor() {
+    super(...arguments);
+  }
```
Overbodig

erikap commented on app/pods/publications/publication/translations/template.hbs:
```diff
@@ -1,23 +1,82 @@
+            />
+          {{else if translationRow.piece}}
+            <Publications::Publication::Translations::TranslationReceivedPanel
+              @translationSubcase={{this.translationSubcase}}
```
Dit argument lijkt niet gebruikt in de component

erikap commented on app/pods/publications/publication/translations/template.hbs:
```diff
@@ -1,23 +1,82 @@
+          {{else if translationRow.piece}}
+            <Publications::Publication::Translations::TranslationReceivedPanel
+              @translationSubcase={{this.translationSubcase}}
+              @piece={{translationRow.piece}}
```
Ik zou verwachten dat het argument voor dit panel een vertaalactiviteit is

erikap commented on app/pods/publications/publication/translations/template.hbs:
```diff
@@ -1,23 +1,82 @@
+
+{{#if this.showTranslationRequestModal}}
+  <Publications::Publication::Translations::TranslationRequestModal
+    @translationSubcase={{this.translationSubcase}}
```
De hele translationSubcase doorgeven om er dan enkel de dueDate van te gebruiken bij initialisatie. Enkel de date doorgeven?

erikap commented on app/pods/publications/publication/translations/controller.js:
```diff
@@ -0,0 +1,197 @@
+
+    yield piece.save();
+    const usedPieces = [piece];
+    console.log(usedPieces)
```
Verwijderen

erikap commented on app/pods/publications/publication/translations/controller.js:
```diff
@@ -0,0 +1,197 @@
+    });
+    yield documentContainer.save();
+
+    const translationActivity = yield requestActivity.translationActivity;
```
Als het model aangepast is en ook translation-activities bevat, kan je de laatste gewoon daaruit halen.

erikap reviewed (changes requested):
De kwaliteit van deze PR moet beter:
- er is nog heel wat legacy die niet opgekuist is na de refactor
- argumenten die niet gebruikt worden, worden meegegeven aan componenten 
- achtergebleven `console.log` statements

De model-hook van de translations route kan ook logischer waardoor de complexiteit in het geheel zal afnemen.

De upload van een vertaaldocument gebeurt nu in 2 stappen, terwijl dit in het design via 1 stap is (file upload bij het formulier).

Commit:
KAS-3106 Undo unused code + simplify inputs for components

Commit:
KAS-3106 Undo console.log

Commit:
KAS-3106 Undo unused arguments

brenner-company committed:
KAS-3106 Update au-kaleidos-css package

brenner-company committed:
KAS-3106 Adjust accordion-panel component to support subtitles

brenner-company committed:
KAS-3106 Fix faulty structure/styling

brenner-company committed:
KAS-3106 Update au-kaleidos-css package (again)

brenner-company committed:
KAS-3106 Adjust panel & accordion-panel component with size argument

brenner-company committed:
KAS-3106 Add size arguments to received- & requested-panels

brenner-company committed:
KAS-3106 Update component styleguide

brenner-company committed:
KAS-3106 Add line to additions file

brenner-company commented:
@TaneeG Structuur en styling zouden nu in orde moeten zijn 👍 Dubbelcheck je nog even?

ValenberghsSven committed:
Merge branch 'development' of github.com:kanselarij-vlaanderen/kaleidos-frontend into feature/KAS-3106_Vertalingen_herwerken

TaneeG committed:
KAS-3106 Remove sourceforprrofpint code

TaneeG committed:
KAS-3106 Reworked routes to an index + made translation received to have translationActivity as param

TaneeG committed:
KAS-3106 Differentiate through array with modelName

TaneeG committed:
Merge branch 'development' into feature/KAS-3106_Vertalingen_herwerken

TaneeG committed:
KAS-3106 Change upload models to multiple file upload

TaneeG committed:
KAS-3106 Remove  difficulty of queues + pushObject to notify disabled

TaneeG committed:
KAS-3106 Make sure translationActivity is visible

TaneeG committed:
KAS-3106 Added design change for files

erikap committed:
Merge branch 'development' into feature/KAS-3106_Vertalingen_herwerken

erikap committed:
Merge branch 'feature/KAS-3106_Vertalingen_herwerken' of github.com:kanselarij-vlaanderen/kaleidos-frontend into feature/KAS-3106_Vertalingen_herwerken

erikap committed:
Fix package-lock.json

erikap commented on app/pods/publications/publication/translations/route.js:
```diff
@@ -1,11 +1,11 @@
   }
 
-  afterModel() {
+  async afterModel() {
```
Waarom deze `async`?

erikap commented on app/pods/publications/publication/translations/index/route.js:
```diff
@@ -1,7 +1,66 @@
+
+export default class PublicationsPublicationTranslationsRoute extends Route {
+  async model() {
+    this.translationSubcase = await this.modelFor('publications.publication.translations');
```
`modelFor` geeft geen promise terug. `await` is overbodig.

erikap commented on app/pods/publications/publication/translations/index/route.js:
```diff
@@ -1,7 +1,66 @@
+  async model() {
+    this.translationSubcase = await this.modelFor('publications.publication.translations');
+
+    let requestActivities = await this.store.query('request-activity',
```
In recente versies moet de `store` service expliciet geinject worden in de route/controller wanneer je ze gebruikt. Zie   https://deprecations.emberjs.com/v3.x/#toc_implicit-injections en deprecation warning in console.

erikap replied:
`const` ipv `let`?

erikap commented on app/pods/publications/publication/translations/index/route.js:
```diff
@@ -1,7 +1,66 @@
+    }
+    return activity;
+  }
+}
```
Waarom deze static `create()` en niet gewoon een constructor? Doorgaans doet @mdv-professional dit wanneer er iets async moet gebeuren, maar hier is er niets async.

Je kan misschien ook werken met een generieke `@tracked activity` en een `isRequest`/`isTranslation`-getter toevoegen die op basis van `this.activity.constructor.modelName` een boolean teruggeeft (zie ook comment op andere PR https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/1196#discussion_r805663255).
`date` zou ook get-function kunnen zijn, wat als voordeel heeft dat die waarde automatisch gaat updaten als de datum van de activity op een andere plaats in de app gewijzigd zou worden.

erikap commented on app/pods/publications/publication/translations/index/route.js:
```diff
@@ -1,7 +1,66 @@
+      }
+    );
+
+    let translationActivities = await this.store.query('translation-activity',
```
`const` ipv `let`?

erikap commented on app/pods/publications/publication/translations/index/route.js:
```diff
@@ -1,7 +1,66 @@
+      }
+    );
+
+    let activities = await Promise.all([ requestActivities.map((request) => Activity.create(request,null)) , translationActivities.map((translation) => Activity.create(null,translation))]);
```
Waarom `await` en `Promise.all`. Hier is toch niets async?

erikap commented on app/pods/publications/publication/translations/index/route.js:
```diff
@@ -1,7 +1,66 @@
+    }
+    if (translationActivity){
+      activity.translationActivity = translationActivity;
+      activity.date = translationActivity.endDate
```
`;` ontbreekt

erikap commented on app/pods/publications/publication/translations/index/route.js:
```diff
@@ -1,7 +1,66 @@
-export default class PublicationsPublicationTranslationsIndexRoute extends Route {
-  beforeModel() {
-    this.transitionTo('publications.publication.translations.documents');
+export class Activity {
```
Iets met `Timeline...` drukt misschien iets beter de intentie van de class uit?  `TimelineEvent` of `TimelineActivity` of zoiets.

erikap commented on app/pods/publications/publication/translations/index/controller.js:
```diff
@@ -0,0 +1,186 @@
+  @task
+  *saveTranslationUpload(translationUpload) {
+    // get latest translation activity
+    const translationActivity = this.model.filter((activity) => !isEmpty(activity.translationActivity))[0].translationActivity;
```
`filter(...)[0]` vervangen door `find()`?

erikap commented on app/pods/publications/publication/translations/index/controller.js:
```diff
@@ -0,0 +1,186 @@
+    // get latest translation activity
+    const translationActivity = this.model.filter((activity) => !isEmpty(activity.translationActivity))[0].translationActivity;
+
+    // triggers call
```
Wat bedoel je met deze comment?

erikap commented on app/pods/publications/publication/translations/index/template.hbs:
```diff
@@ -0,0 +1,72 @@
+                @skin="secondary"
+                @icon="plus"
+                @disabled={{this.isTranslationUploadDisabled}}
+                {{on "click" this.openTranslationUploadModal }}
```
Spatie weg aan het eind

erikap commented on app/pods/publications/publication/translations/index/controller.js:
```diff
@@ -0,0 +1,186 @@
+  }
+
+  get isTranslationUploadDisabled() {
+    return this.model.length === 0;
```
In de praktijk zal dit werken omdat request- en translation-activity altijd in paren voorkomen in de huidige app. Maar wat je hier logisch wil uitdrukken is dat er minstens een translation-activity in het model zit (waar je die uploaded translation dan aan kan koppelen).

Hier zou ik misschien zelfs een getter voor durven maken `latestTranslationActivity` die je dan ook kan hergebruiken in de upload task.

erikap commented on app/pods/publications/publication/translations/index/controller.js:
```diff
@@ -0,0 +1,186 @@
+        translationUpload.receivedAtDate
+      );
+
+      this.publicationSubcase.endDate = translationUpload.receivedAtDate;
```
Is het niet de `endDate` van de translation-subcase die we hier willen zetten?
Zo ja, is `this.publicationSubcase` niet nodig in deze controller.

erikap commented on app/pods/publications/publication/translations/index/controller.js:
```diff
@@ -0,0 +1,186 @@
+      !this.translationSubcase.receivedDate
+    ) {
+      this.translationSubcase.receivedDate = translationUpload.receivedAtDate;
+      yield this.translationSubcase.save();
```
Save promise kan bij in de `Promise.all`?

erikap commented on app/pods/publications/publication/translations/index/controller.js:
```diff
@@ -0,0 +1,186 @@
+
+      piece.receivedDate = translationUpload.receivedAtDate;
+      piece.language = language;
+      piece.translationActivityGeneratedBy= translationActivity;
```
Spatie voor `=`

erikap commented on app/components/publications/publication/translations/translation-upload-modal.js:
```diff
@@ -1,43 +1,41 @@
-  @tracked file;
-  @tracked name;
-  @tracked isSourceForProofPrint = false;
+  @tracked generatedPieces = [];
```
`uploadedPieces` misschien duidelijker in deze context?

erikap commented on app/components/publications/publication/translations/translation-upload-modal.js:
```diff
@@ -1,43 +1,41 @@
-  @tracked isSourceForProofPrint = false;
+  @tracked generatedPieces = [];
   @tracked receivedAtDate = new Date();
   @tracked isTranslationIn = false;
```
Betekenis van deze property was mij zonder de schermen niet duidelijk. `mustUpdatePublicationStatus` of zoiets?

erikap commented on app/components/publications/publication/translations/translation-upload-modal.js:
```diff
@@ -49,31 +47,25 @@ export default class PublicationsTranslationTranslationUploadModalComponent exte
+    if (this.generatedPieces.length > 0) {
+      for (let piece of this.generatedPieces) {
+        yield this.deleteUploadedPiece.perform(piece);
+      }
```
Optimalisatie: parallelliseren met `Promise.all`

erikap commented on app/components/publications/publication/translations/translation-upload-modal.js:
```diff
@@ -49,31 +47,25 @@ export default class PublicationsTranslationTranslationUploadModalComponent exte
     yield this.args.onSave({
-      file: this.file,
-      name: this.name,
+      generatedPieces: this.generatedPieces,
```
Zelfde comment als voor de tracked properties. `uploadedPieces` en `mustUpdatePublicationStatus` misschien duidelijker in deze context?

erikap commented on app/components/publications/publication/translations/translation-upload-modal.js:
```diff
@@ -1,43 +1,41 @@
+    this.generatedPieces.pushObject(piece);
   }
 
   @task({
```
Ter info: je kan hier ook `@dropTask` gebruiken.

erikap commented on app/components/publications/publication/translations/translation-upload-modal.js:
```diff
@@ -86,10 +78,12 @@ export default class PublicationsTranslationTranslationUploadModalComponent exte
+    const file = yield piece.file;
+    yield file.destroyRecord();
+    const documentContainer = yield piece.documentContainer;
+    yield documentContainer.destroyRecord();
```
Optimalisatie: parallelliseren met `Promise.all`

erikap commented on app/components/publications/publication/translations/translation-request-modal.hbs:
```diff
@@ -25,6 +22,48 @@
+            </div>
+          </div>
+        </div>
         {{! template-lint-enable no-inline-styles }}
```
Kan dit al hoger in de template gezet worden?

erikap commented on app/components/publications/publication/translations/translation-request-modal.hbs:
```diff
@@ -49,26 +88,30 @@
             @value={{mut this.message}}
           />
         </div>
-        {{! Attachments }}
```
Kunnen we hier consequent in zijn. Ofwel bij geen enkel veld die comment zetten ofwel overal. Mij maakt het niet uit wat van beide het wordt.

erikap commented on app/components/publications/publication/translations/translation-request-modal.hbs:
```diff
@@ -49,26 +88,30 @@
+          </Auk::Label>
+          <Auk::FileUploader
+            @multiple={{true}}
+            @onUpload={{this.uploadPiece}}
```
Na het uploaden van files wordt dit gewoon een groene kader.  Er is geen lijst van de bestanden die opgeladen zijn en er kunnen ook geen extra bestanden meer upgeloaden worden. Dat lijkt me niet duidelijk voor de gebruikers.

TaneeG replied:
Hiervoor is een nieuw ticket: https://kanselarij.atlassian.net/browse/KAS-3194

erikap commented on app/components/publications/publication/translations/translation-request-modal.hbs:
```diff
@@ -49,26 +88,30 @@
+      {{on "click" (perform this.save)}}
     >
-      {{t "request-action"}}
+      {{t "save"}}
```
In prototype is dit 'Aanvragen'

erikap commented on app/components/publications/publication/translations/translation-request-modal.js:
```diff
@@ -4,42 +4,73 @@ import { inject as service } from '@ember/service';
   @service store;
 
-  @tracked translationDueDate = this.args.translationSubcase.dueDate ? this.args.translationSubcase.dueDate : new Date();
+  @tracked usedPieces = [];
```
`uploadedPieces` is duidelijker in  deze context?

erikap commented on app/components/publications/publication/translations/translation-request-modal.js:
```diff
@@ -4,42 +4,73 @@ import { inject as service } from '@ember/service';
+  @task
+  *save() {
+    yield this.args.onSave({
+      usedPieces: this.usedPieces,
```
`uploadedPieces` is duidelijker in deze context?

erikap commented on app/components/publications/publication/translations/translation-request-modal.js:
```diff
@@ -4,42 +4,73 @@ import { inject as service } from '@ember/service';
+    yield file.destroyRecord();
+    const documentContainer = yield piece.documentContainer;
+    yield documentContainer.destroyRecord();
+    yield piece.destroyRecord();
```
Zelfde comments als in translation-upload-model over `dropTask` en optimalisaties dmv parallellisatie.

erikap commented on app/components/publications/publication/translations/translation-request-modal.js:
```diff
@@ -4,42 +4,73 @@ import { inject as service } from '@ember/service';
 
   constructor() {
     super(...arguments);
-    this.setEmailFields.perform();
```
Moet `this.subject` en `this.message` niet geinitialiseerd worden?

TaneeG replied:
Ik dacht hier het eerder te laten komen vanaf er iets werd ingevuld.

erikap commented on app/pods/publications/publication/translations/index/controller.js:
```diff
@@ -0,0 +1,186 @@
+    for (let piece of translationRequest.usedPieces){
+      piece.translationSubcaseSourceFor = this.translationSubcase;
+      const documentContainer = yield piece.documentContainer;
+      yield documentContainer.save();
```
Dit zou ik in de modal al doen. De document-container wordt daar immers ook gedestroyed als de modal afgesloten wordt zonder opslaan.

erikap commented on app/components/publications/publication/translations/translation-upload-modal.js:
```diff
@@ -1,43 +1,41 @@
-    this.name = file.filenameWithoutExtension;
+  uploadPiece(file) {
+    const now = new Date();
+    const documentContainer = this.store.createRecord('document-container', {
+      created: now,
+    });
```
Deze document-container wordt nergens gesaved.

erikap commented on app/pods/publications/publication/translations/index/controller.js:
```diff
@@ -0,0 +1,186 @@
+      piece.language = yield this.store.findRecordByUri('language', CONSTANTS.LANGUAGES.NL);
+
+      yield piece.save();
+      usedPieces.push(piece);
```
Waarom deze extra `usedPieces` array? Je kan toch overal `translationRequest.usedPieces` gebruiken?

erikap commented on app/pods/publications/publication/translations/index/controller.js:
```diff
@@ -0,0 +1,186 @@
+      const documentContainer = yield piece.documentContainer;
+      yield documentContainer.save();
+      piece.pages = translationRequest.pagesAmount;
+      piece.words = translationRequest.wordsAmount;
```
Dit is niet correct. Aantal blz. en woorden die ze invullen in de modal is de som van alle documenten. We weten het aantal blz./woorden dus niet per document. Deze velden mogen leeg gelaten worden.

erikap commented on app/pods/publications/publication/translations/index/controller.js:
```diff
@@ -0,0 +1,186 @@
+    });
+    yield requestActivity.save();
+
+    const french = yield this.store.findRecordByUri('language', CONSTANTS.LANGUAGES.FR);
```
Waarom een aparte variable en niet inline zoals bij NL?

erikap commented on app/pods/publications/publication/translations/index/controller.js:
```diff
@@ -0,0 +1,186 @@
+
+    const outboxPromise = this.store.findRecordByUri('mail-folder', PUBLICATION_EMAIL.OUTBOX);
+    const mailSettingsPromise = this.store.queryOne('email-notification-setting');
+    const [files,outbox, mailSettings] = yield Promise.all([filesPromise,outboxPromise, mailSettingsPromise]);
```
Spaties na `,`

erikap replied:
Qua leesbaarheid zou ik dit allemaal inline doen
```
const [files, outbox, mailSetting] = yield Promise.all([
    ...usedPieces.mapBy('file')),
    this.store.record.findRecordByUri(...),
    ...
]);

erikap commented on app/pods/publications/publication/translations/index/controller.js:
```diff
@@ -0,0 +1,186 @@
+      this.publicationFlow,
+      CONSTANTS.PUBLICATION_STATUSES.TO_TRANSLATIONS
+    );
+    yield pubStatusChange;
```
Waarom `yield` niet inline?

erikap commented on app/pods/publications/publication/translations/index/controller.js:
```diff
@@ -0,0 +1,186 @@
+
+  @task
+  *deleteRequest(requestActivity){
+    const saves = [];
```
`saves` is hier een vreemde name voor promises die verwijderen.

erikap commented on app/components/publications/publication/translations/translation-requested-panel.js:
```diff
@@ -0,0 +1,29 @@
+ * @argument {requestActivity}
+ */
+export default class PublicationsPublicationTranslationTranslationRequestedPanel extends Component {
+  @service store;
```
Wordt niet gebruikt?

erikap commented on app/components/publications/publication/translations/translation-requested-panel.hbs:
```diff
@@ -0,0 +1,80 @@
+          <h4 class="auk-u-m-0">{{t "message"}}</h4>
+          {{! TODO: Remove inline style }}
+          {{! template-lint-disable no-inline-styles }}
+          {{! p-tag opening and closing on 1 line to prevent whitespace to be rendered. }}
+          <p style="white-space:pre-line;">
+            {{@requestActivity.email.message}}
+          </p>
```
Ofwel moet de comment weg, ofwel moet de paragraph weer op 1 lijn.

ValenberghsSven replied:
Ik zie in huidige code nog plaatsen waar diezelfde comment stond en ook linter op gerund is in andere PR's
(proofs/requests/template, translations/requests/template), mogelijk apart te bekijken.

Gebruik `{{! prettier-ignore }}` zodat prettier die ene regel niet formateerd bij autoformat.

erikap commented on app/components/publications/publication/translations/translation-requested-panel.hbs:
```diff
@@ -0,0 +1,80 @@
+            @skin="borderless"
+            @icon="three-dots"
+            as |Menu|>
+            <Menu.Item
+              @skin="danger"
+              {{on "click" this.promptDeleteRequest}}
+            >
+              {{t "delete-translation-request"}}
+            </Menu.Item>
```
Ik zou delete niet toestaan als de translation-activity die bij de aanvraag hoort een einddatum heeft, want d.w.z. dat er al vertaaldocumenten opgeladen zijn voor de aanvraag.

TaneeG committed:
KAS-3106 prettier everywhere

TaneeG committed:
KAS-3106 optimize modals & panels

TaneeG committed:
KAS-3106 optimize controller

TaneeG committed:
KAS-3106 optimize route

TaneeG committed:
KAS-3106 optimize route & controller

TaneeG committed:
KAS-3106 optimize route & controller

TaneeG committed:
KAS-3106 remove await & promise for making timeline

MikiDi commented on app/components/publications/publication/translations/translation-request-modal.hbs:
```diff
@@ -1,74 +1,106 @@
-            @onChange={{this.setTranslationDueDate}}
-          />
-        </div>
-        {{! template-lint-enable no-inline-styles }}
```
`{{! template-lint-enable no-inline-styles }}`
Deze mag wel blijven staan. Andere, minder bewust ingevoegde inline styles worden dan nog steeds opgemerkt. Zelfde voor `translation-requested-panel.hbs`.

brenner-company replied:
Ik ga in de nabije toekomst hier even kort een item voor opnemen rond:
- al dan niet verschil tussen `form-group` en `input-group` (laatste zal misschien verdwijnen aangezien die niets doet).
- mogelijks toe voegen van width modifier classes (in de aard van `.auk-form-group--w-medium`) om zo inline styling te vermijden.

Indien die laatste doorgevoerd wordt doorloop ik sowieso nog eens alle gevallen van `class="auk-form-group"` waar er inline styling aan te pas komt.

TaneeG committed:
KAS-3106 don't show delete request

TaneeG committed:
KAS-3106 show uploaded files in modals

TaneeG committed:
KAS-3106 return correct latest activity

erikap committed:
Make date getter of timeline-activity dynamic

Return date property depending on the type of activity

erikap committed:
Simplify array merging in model-hook

erikap committed:
Remove void async from afterModel hook

erikap committed:
Add explicit injection of store in route

erikap committed:
Remove disabled state based on end-date of translation-subcase

UI needs to be as flexible as possible towards the user. They should
be able to keep changing all things as long as the publication has not
been published in BS.

erikap committed:
Replace isEmpty with null check for an object

erikap committed:
Sort timeline activities on start-date

By sorting on start-date we ensure request- and translation-activities
always appear in a sequence next to each other.

erikap committed:
Simplify getter to retrieve latest translation-activity

Since the model is already reverse sorted on start-date, we know the
first translation-activity found in the array is the most recent one.

erikap committed:
Fetch language once outside for-loop

erikap committed:
Fix casing of timeline-activity

TimelineActivity instead of TimeLineActivity

erikap committed:
Fetch language once outside for-loop

erikap committed:
Save uploaded pieces in parallel

erikap committed:
Remove void yield on store.createRecord

erikap committed:
Rename language var

erikap committed:
Remove void yield on store.createRecord

erikap committed:
Revert multi file upload

Auk::FileUploader yields an error when uploading multiple files at the
same time. Will be fixed in another PR.

erikap committed:
Simplify parallellisation of file deletions in request- and
translation upload modal

erikap committed:
Remove unused properties from translation-request onSave hook

erikap committed:
Update translation-request-modal documentation

erikap committed:
Update translation-subcase due date in save-hook of translation-request

erikap committed:
Add condition to check whether mail exists on removal of request-activity

erikap committed:
Fix update of number of pages/words in request mail preview

erikap committed:
Mark all required fields in translation-request modal

erikap committed:
Replace if-not with unless in template

erikap committed:
Merge branch 'development' into feature/KAS-3106_Vertalingen_herwerken

erikap reviewed (approved):
Ik heb nog wat verbeteringen/fixes gedaan @TaneeG . Overloop die zeker eens.

Er waren verschillende comments uit de vorige review gemarkeerd als 'Resolved' terwijl die niet opgelost waren. Of nieuwe issues geintroduceerd (vb. als je een `await` verwijderd; kijk dan ook na of je de `async` van een method moet halen). Dat maakt het review proces heel omslachtig en tijdrovend.

MikiDi committed:
don't use droptask for a task which needs all invocations executed

TaneeG commented on app/components/publications/publication/translations/translation-upload-modal.hbs:
```diff
@@ -1,88 +1,41 @@
+          <div class="auk-o-flex auk-o-flex--horizontal auk-o-flex--spaced">
+            <Auk::Badge @icon="paperclip" @size="regular" />
+            <h5 class="auk-h4 auk-u-m-0">
+              {{piece.file.downloadFilename}}
```
In het onterp moet ook de file extension worden gedisplayed

erikap committed:
Fix parallel piece save promises on translation upload

erikap commented on app/pods/publications/publication/translations/index/controller.js:
```diff
@@ -0,0 +1,193 @@
+    if (translationUpload.mustUpdatePublicationStatus) {
+      yield this.publicationService.updatePublicationStatus(
+        this.publicationFlow,
+        CONSTANTS.PUBLICATION_STATUSES.TRANSLATION_IN,
```
Ik vind dat persoonlijk ook duidelijker. Te checken met het team en kan dan misschien meegenomen worden in KAS-3188.

erikap commented on app/pods/publications/publication/translations/index/controller.js:
```diff
@@ -0,0 +1,193 @@
+      pieceSaves.push(piece.save());
+    }
+
+    translationActivity.endDate = translationUpload.receivedAtDate;
```
De `translation-subcase.end-date` wordt aangepast als er expliciet aangeduid is dat de status gewijzigd moet worden. Je hebt gelijk dat er dan cases gaan zijn zoals diegene  die jij beschrijft, waar de datums in een onlogische volgorde gaan staan. Maar dat zijn de uitzonderingen. Ik zou niet focussen om al die cases te coveren nu. Doorgaans zal het proces gewoon logisch verlopen: 1 aanvraag, 1 antwoord waarbij translation files geupload worden en wijzigen naar volgende status.

erikap commented on app/components/publications/publication/translations/translation-upload-modal.hbs:
```diff
@@ -1,88 +1,41 @@
+      />
+    </div>
+    <div class="auk-form-group">
+      <Auk::Label for="translationUploadReceivedAtDate">
```
Op validations zou ik momenteel ook niet te veel focussen. Hannes heeft al een paar keer aangehaald dat we de gebruikers zoveel mogelijk vrijheid moeten geven. Verplichte velden, tenzij expliciet nodig voor een goede werking,  werken alleen belemmerend.
Het weergeven van errors is waarschijnlijk wel nog iets dat naar de toekomst toe consistenter moet zijn en verwerkt in de auk-componenten. Maar in deze PR zou ik er dus niets aan doen.

erikap commented on app/components/publications/publication/translations/translation-request-modal.hbs:
```diff
@@ -1,74 +1,120 @@
+            <div class="auk-o-grid-col-6">
+              <div class="auk-form-group">
+                <Auk::Label for="translationLoadPages" class="auk-label--light">
+                  {{t "pages-amount"}}
```
De bemerking dat hun bericht overschreven wordt als ze een van de invulvelden wijzigen, is in de vorige iteratie ook aan gehaald en als non-issue geklasseerd. In de praktijk vullen ze de invoervelden in en gaan ze in de gegenereerde mail message zelden iets wijzigen.
Aantal pagina's en woorden wordt nu niet meer in de backend opgeslagen, dus in dat opzicht zijn de velden inderdaad overbodig, maar om consequent te zijn met de andere datapunten, zou ik ze toch als invulveld houden ipv dat ze die rechtstreeks in de mail zouden moeten gaan aanpassen.

erikap commented on app/components/publications/publication/translations/translation-request-modal.js:
```diff
@@ -2,59 +2,91 @@ import Component from '@glimmer/component';
-      totalPages: this.totalPages,
-      totalWords: this.totalWords,
-      totalDocuments: this.totalDocuments,
+      totalPages: this.pagesAmount,
```
Terechte opmerking, maar moeilijk te verbeteren nu met zoveel branches in progress zonder conflicten te creëeren. Ik stel voor om hier een story voor te maken die pas opgenomen wordt als de stories voor vertaling/drukproeven/publicaties gemerged zijn.

erikap commented on app/components/publications/publication/translations/translation-request-modal.hbs:
```diff
@@ -1,74 +1,120 @@
+        <Auk::List>
+          {{#each this.uploadedPieces as |piece|}}
+            <Auk::List::Item
+              class="auk-o-flex auk-o-flex--horizontal auk-o-flex--spaced-wide"
```
Ik stel voor om het hier te laten doorgaan. Je mag in je publicatie-branch met `Utils::Documents::List` proberen en als dat beter is, deze template gelijk te trekken.

ValenberghsSven replied:
die `auk-o-flex--horizontal` bestaat zelfs niet, `auk-o-flex--horizontal-center` wel, maar dat is niet wat je hier wil.

Dit staat ook zo op de usertest app, mogelijk komt dit daarvan (daar doet het ook niks)

erikap committed:
Merge branch 'feature/KAS-3106_Vertalingen_herwerken' of github.com:kanselarij-vlaanderen/kaleidos-frontend into feature/KAS-3106_Vertalingen_herwerken

ValenberghsSven commented on app/components/publications/publication/translations/translation-request-modal.js:
```diff
@@ -2,59 +2,91 @@ import Component from '@glimmer/component';
+    this.args.onCancel();
+  }
+
+  @dropTask
```
Michaël heeft 1 voorval van een `@dropTask` weggehaald, maar deze mag hier ook niet staan?
Het gebruik van droptask zou ik eerder gebruiken bij data loading, minder bij creates/deletes.
Ook lijkt dit niet helemaal te werken in combinatie met "@cancelDisabled={{this.isCancelDisabled}}" op de footer.
Met de droptasks is die knop niet disabled en kan je meerdere malen klikken op annuleren, waardoor de tasks niet goed worden afgehandeld en  de modal gewoon blijft openstaan met nog docs erin.
Met gewone tasks werkt die disabled wel, en kan je maar 1 keer klikken, dus de task wordt maar 1 keer uitgevoerd en alles is netjes verwijderd

Ik zou de PR er niet op tegenhouden, gewoon later even terug naar kijken (na upload multiple true fix)

ValenberghsSven committed:
prettier-ignore must be directly above the line to be ignored

ValenberghsSven committed:
Merge branch 'feature/KAS-3106_Vertalingen_herwerken' of github.com:kanselarij-vlaanderen/kaleidos-frontend into feature/KAS-3106_Vertalingen_herwerken

ValenberghsSven commented on app/components/publications/publication/translations/translation-request-modal.js:
```diff
@@ -68,13 +100,33 @@ export default class PublicationsTranslationRequestModalComponent extends Compon
+      created: now,
+    });
+    await documentContainer.save();
+    const piece = this.store.createRecord('piece', {
+      created: now,
+      modified: now,
+      file: file,
```
Eerder een algemene opmerking bij publicatie pieces.
We zetten nergens een accessLevel bij, mogelijk omdat we dat niet echt nodig hebben voor ovrb.
Echter, bij een aanpassing van de piece gaan we sowieso in de override van de save (piece.js model) de waarde van `accessLevelLastModified` invullen. Dat strookt niet helemaal

![image](https://user-images.githubusercontent.com/22245223/155004227-1214c8f3-52f3-4d5d-bea4-5c1ddea2951f.png)

ValenberghsSven commented on app/components/publications/publication/translations/translation-request-modal.js:
```diff
@@ -2,59 +2,91 @@ import Component from '@glimmer/component';
+  }
+
+  @dropTask
+  *deleteUploadedPiece(piece) {
+    const file = yield piece.file;
+    const documentContainer = yield piece.documentContainer;
+    yield Promise.all([
+      file.destroyRecord(),
+      documentContainer.destroyRecord(),
+    ]);
+    this.uploadedPieces.removeObject(piece);
```
1- Hier wordt piece zelf niet verwijderd, model is nog niet gesaved maar een cleanup is snel gebeurd.

2- Dit is een lastige qua code inconsistentie.
Op 6 andere plaatsen in onze code doen we dit process (file, container en piece verwijderen)
Op 3 van die plaatsen (agenda/agendaitem/subcase is de code 100% duplicate (mogelijkheid tot util of service)
De andere 3 plaatsen (+ deze dus 4) zitten in publicatie en ze zijn allemaal anders.
Er wordt meer met promises gewerkt, maar telkens op een andere manier. Die droptask erbij maakt het hier nog verwarrender.
Ook is er de fileService, die ook gelijkaardig dingen kan opkuisen (maar heeft nood aan een likje "task" verf.)

**Moet niet in deze PR gelijk getrokken worden,** ik merk het gewoon op.
een voorbeeld van zo'n duplicate
https://github.com/kanselarij-vlaanderen/frontend-kaleidos/blob/feature/KAS-3106_Vertalingen_herwerken/app/pods/agenda/agendaitems/agendaitem/documents/controller.js#L126

ValenberghsSven commented on app/components/publications/publication/translations/translation-requested-panel.hbs:
```diff
@@ -0,0 +1,86 @@
+  </Auk::AccordionPanel>
+</div>
+
+{{#if this.isVerifyingDelete}}
+  <WebComponents::VlModalVerify
+    @title={{t "delete-translation-request"}}
+    @message={{t "delete-translation-request-message"}}
+    @cancel={{this.cancelDeleteRequest}}
+    @verify={{this.deleteRequest}}
+  />
+{{/if}}
```
Gebruik van VL componenten vermijden zodat die uiteindelijk eruit kunnen worden gehaald, kan makkelijk vervangen worden met auk componenten.

ValenberghsSven commented on app/components/publications/publication/translations/translation-requested-panel.js:
```diff
@@ -0,0 +1,43 @@
+  }
+
+  @action
+  promptDeleteRequest() {
```
Waarom "prompt" en niet "verify" of "show" of "open".
Van deze 4 gebruiken we meestal "open", "prompt" slechts 2 keer + deze

ValenberghsSven commented on app/pods/publications/publication/translations/index/controller.js:
```diff
@@ -0,0 +1,193 @@
+    this.showTranslationRequestModal = false;
+  }
+
+  @dropTask
+  *deleteRequest(requestActivity) {
+    const deletePromises = [];
+
+    const translationActivity = yield requestActivity.translationActivity;
+    deletePromises.push(translationActivity.destroyRecord());
+
+    const mail = yield requestActivity.email;
+    if (mail) {
+      deletePromises.push(mail.destroyRecord());
+    }
+
+    deletePromises.push(requestActivity.destroyRecord());
+
+    const pieces = yield requestActivity.usedPieces;
+
+    for (const piece of pieces.toArray()) {
+      const [file, documentContainer] = yield Promise.all([
+        piece.file,
+        piece.documentContainer,
+      ]);
+
+      deletePromises.push(piece.destroyRecord());
+      deletePromises.push(file.destroyRecord());
+      deletePromises.push(documentContainer.destroyRecord());
+    }
+    yield Promise.all(deletePromises);
+    this.send('refresh');
+  }
```
code gooit errors, DELETE naar `request-activities/id` faalt
{
  "code": "500",
  "status": "Server Error",
  "title": "Could not execute SPARQL query."
}

ValenberghsSven reviewed (commented):
Ik had iets moeten zeggen, maar was aan het reviewen toen dit gemerged werd :sweat_smile: 
Ik rond dan maar af..
